File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        double cont = 1; // all ones
        for(int i = 1; i < n && (n-i) >= i; i++){
            System.out.println(i);
            cont+=(factorial(n-i, i)/factorial(i,i));
        }
        return (int)cont;
    }
    
    private double factorial(long n, int stop){
        if(n == 0 || stop <= 0){
            return 1;
        }
        if(n == 1 || stop <= 0){
            return 1;
        }
        return n*factorial(n-1, stop-1);
    }
}

File: 70-climbing-stairs/70-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        double ways = 0;
        for(int i = 0; i < n && (i <= n-i); i++){
            ways+=factorial(n-i, i)/factorial(i,i);
        }
        return (int) ways;
    }
    double factorial(int n, int stop){
        if(n <= 0 or n <= 1 or stop <= 0){
            return 1;
        }
        return n*(factorial(n-1, stop-1));
    }
};